ql/patterns/observable.hpp

```diff
@@ -264,14 +264,15 @@ namespace QuantLib {
 #include <boost/enable_shared_from_this.hpp>
```

Comment:
I think this should go in `ql/shared_ptr.hpp`, too?  (If the `using` statement there works using the boost classes, instead, that means we can remove this here.)